.github/workflows/pypi-cd.yml

```diff
@@ -198,31 +198,64 @@ jobs:
                   path: python/wheels
                   if-no-files-found: error
 
-    publish-to-pypi:
+    build-source-dist-and-publish-to-pypi:
         if: github.event_name != 'pull_request'
         name: Publish the base PyPi package
         runs-on: ubuntu-latest
         needs: publish-binaries
         steps:
-            - uses: actions/download-artifact@v4
+            - name: Checkout
+              uses: actions/checkout@v4
+              with:
+                  submodules: "true"
+
+            - name: Set the release version
+              shell: bash
+              run: |
+                  if ${{ env.EVENT_NAME == 'pull_request' }}; then
+                    R_VERSION="255.255.255"
+                  elif ${{ env.EVENT_NAME == 'workflow_dispatch' }}; then
+                    R_VERSION="${{ env.INPUT_VERSION }}"
+                  else
+                    R_VERSION=${GITHUB_REF:11}
+                  fi
+                  echo "RELEASE_VERSION=${R_VERSION}" >> $GITHUB_ENV
+              env:
+                  EVENT_NAME: ${{ github.event_name }}
+                  INPUT_VERSION: ${{ github.event.inputs.version }}
+
+            - name: Set the package version for Python
+              working-directory: ./python
+              run: |
+                  sed -i "s|255.255.255|${{ env.RELEASE_VERSION }}|g" ./Cargo.toml
```

Comment:
This is already done in another step, don't duplicate. You can re-use the results.